src/ansys/fluent/core/utils/search.py

```diff
@@ -212,10 +238,320 @@ def inner(tree, path, root_path):
+
+    Parameters
+    ----------
+    search_string: str
+        The word to search for. Semantic search is default.
+    language: str
+        The language for the semantic search.
+        English is default for the semantic search.
+        ISO 639-3 code of the language to be used for semantic search.
+        See `https://omwn.org/omw1.html` for the list of supported languages.
+        The default value is `eng` for English language.
+
+    Returns
+    -------
+    queries: list
+        List of search string matches.
+    """
+    names = _get_api_object_names()
+    similar_keys = set()
+    search_string_synsets = (
+        wn.synsets(search_string.decode("utf-8"), lang=language)
+        if sys.version_info[0] < 3
+        else wn.synsets(search_string, lang=language)
+    )
+    for name in names:
+        api_object_name_synsets = (
+            wn.synsets(name.decode("utf-8"), lang=language)
+            if sys.version_info[0] < 3
+            else wn.synsets(name, lang="eng")
+        )
+        for search_string_synset in search_string_synsets:
+            for api_object_name_synset in api_object_name_synsets:
+                search_string_synset_name = search_string_synset.name().split(".")[0]
+                api_object_synset_name = api_object_name_synset.name().split(".")[0]
+                if (
+                    search_string in api_object_synset_name
+                    or search_string_synset_name in api_object_synset_name
+                ):
+                    similar_keys.add(api_object_synset_name + "*")
+    if similar_keys:
+        for key in similar_keys:
+            _search_wildcard(key)
+    else:
+        queries = _get_close_matches_for_word_from_names(search_string, names)
+        if queries:
+            _print_search_results(queries, wildcard=True)
+
+
+def search(
+    search_string: str,
+    language: Optional[str] = "eng",
+    wildcard: Optional[bool] = False,
+    match_whole_word: Optional[bool] = False,
+    match_case: Optional[bool] = False,
+):
+    """Search for a word through the Fluent's object hierarchy.
+
+    Parameters
+    ----------
+    search_string: str
+        The word to search for. Semantic search is default.
+    language: str
+        The language for the semantic search.
+        English is default for the semantic search.
+        ISO 639-3 code of the language to be used for semantic search.
+        See `https://omwn.org/omw1.html` for the list of supported languages.
+        The default value is `eng` for English language.
+    wildcard: bool
+        Whether to use wildcard pattern. If ``True`` will match wildcard pattern based on ``fnmatch`` module and
+        will turn off semantic matching.
+    match_whole_word: bool
+        Whether to get exact match. If ``True`` will match exact string and will turn off semantic matching.
+    match_case: bool
+        Whether to match case. If ``True`` will match case-insensitive case.
```

Comment:
```suggestion
    wildcard: bool, optional
        Whether to use the wildcard pattern. The default is ``False``. If ``True``, the
        wildcard pattern is based on the ``fnmatch`` module and semantic matching
        is turned off.
    match_whole_word: bool, optional
        Whether to find only exact matches. The default is ``False``. If ``True``,
        only exact matches are found and semantic matching is turned off.
    match_case: bool, optional
        Whether to match case. The default is ``False``. If ``True``, the search is case-insensitive.
```
After a parameter's data type, the practice is to add a comma and "optional" if there is a default value. The default value is then noted immediately after the brief description. Another option is to follow the parameter's data type with a comma and then "default: X", where X is the default value. When this method is used, the default doesn't have to be given after the brief description.